client/src/App.js

```diff
@@ -1,12 +1,210 @@
-import React, {useEffect, useState} from 'react';
-import logo from './logo.svg';
-import Ticket from './components/Ticket'
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import icon from './favicon.ico';
+import Ticket from './components/Ticket';
+import Search from './components/Search';
+import Sidebar from './components/Sidebar';
+import ShowButton from './components/ShowButton';
 import './App.css';
 
 function App() {
+  const [options, setOptions] = useState({
+    ThemeColor: { color: '#00a4a4' },
+    displayMenu: false,
+    filterLabels: [],
+    hideDone: { active: false },
+    timeRange: { active: false, range: 'Always' },
+  });
+  const [ticketsToDisplay, setTicketsToDisplay] = useState(['loading']);
+  const getAvailableLabels = (tickets) => {
+    const labelsArray = [];
+    tickets.forEach((ticket) => {
+      if (!ticket.labels) { return; }
+      ticket.labels.forEach((label) => {
+        if (!labelsArray.includes(label))labelsArray.push(label);
+      });
+    });
+    console.log('Available Labels', labelsArray);
+    const availableLabels = labelsArray.map((label) => Object({ name: label, active: false }));
+    const optionsWithLabels = { ...options };
+    optionsWithLabels.filterLabels = availableLabels;
+    setOptions(optionsWithLabels);
+  };
+  async function grabTickets() {
+    const tickets = (await axios.get('/api/tickets')).data;
+    console.log('brought', tickets);
+    getAvailableLabels(tickets);
+    setTicketsToDisplay(tickets);
+  }
+  useEffect(() => {
+    grabTickets();
+  }, []);
+  const toggleMenu = () => {
+    const changedOptions = { ...options };
+    changedOptions.displayMenu = !changedOptions.displayMenu;
+    setOptions(changedOptions);
```

Comment:
When set state depends on old state better use callback

Reply:
what do you mean? use a callback for what?